artiq/coredevice/suservo.py

```diff
@@ -284,7 +284,7 @@ def set_dds_mu(self, profile, ftw, offs, pow_=0):
         """
         base = (self.servo_channel << 8) | (profile << 3)
         self.servo.write(base + 0, ftw >> 16)
-        self.servo.write(base + 6, ftw)
+        self.servo.write(base + 6, (ftw & 0xffff))
```

Comment:
With the current master, the docstring for `get_profile_mu` is not correct, since the lower half of `ftw` is written to/read from the coefficient memory as a `w.coeff` (18-bit) number so https://github.com/m-labs/artiq/blob/704b5c63054fe6d62b2afd88f91708af0e88eea6/artiq/coredevice/suservo.py#L432 should be `ftw & 0x3ffff`.

We could fix this by either modifying the docs or by masking the ftw off when writing.

We chose the latter approach here on the basis that all entries in the state/coefficient memories are either: signed with a width of `w.coeff` *or* unsigned with a width < `w.coeff`. This allows us to sign extend the RTIO interface uniformly without any special casing. If we store the lower half of `ftw` as a `w.coeff` number then this is no longer true, and `ftw[w.coeff-1]' is incorrectly interpreted as a sign bit. So long as this is correctly documented, it's not a problem (users can mask the bits off when reconstructing the `ftw`) but it seemed like a slightly ugly special case to have to document so I preferred to just apply the bit mask during writes.